lib/middleware/httpRequestAllowed.ts

```diff
@@ -8,9 +8,19 @@ const isRequestAllowed = (
   return async (req: NextApiRequest, res: NextApiResponse): Promise<unknown> => {
     try {
       const session = await getSession({ req });
-      const requestBody = JSON.parse(req.body);
-      if (requestBody.method && methods.includes(requestBody.method)) {
-        switch (requestBody.method) {
+      let method;
+      let isAllow = false;
```

Comment:
You could write this entire thing in way fewer lines 

```JS
const requestBody = JSON.parse(req.body);
const method =  requestBody.method && methods.includes(requestBody.method) ? requestBody.method : req.method
switch(method){
  ...
}
```

Also if you're changing this file tests need to be reevaluated as well to reflect new code paths

Reply:
Yeah i know but you may have forget that  there's  no requestBody in the context of bearer Api. Meaning we do not send any data like template does. So requestBody will be undefined.

Reply:
In that case you can do this 

```JS
const requestBody = req.body ? JSON.parse(req.body): undefined;
const method =  requestBody?.method && methods.includes(requestBody.method) ? requestBody.method : req.method
switch(method){
  ...
}
```

Accounts for no req.body and still way fewer lines :)